### PR TITLE
Improve logging to handle more scenarios and print more information. …

### PR DESCRIPTION
**Pull request by harry8525**
…Fix a few random bugs.

### CODE REVIEW

**Comment by dzearing** [comment]
When i created 1 failure, i got 6 failures in the output. I think we should fix that first. Maybe we could split this merge into 2... the typings and such as totally fine and great, and there are quite a few bug fixes here but the logging is busted

File: src/GulpTask.ts [hunk 1]
```diff
@@ -68,28 +77,37 @@ export class GulpTask<TASK_CONFIG> implements IExecutable {
...
-          .on('queueDrain', () => {
-            resolve();
-          })
-          .on('end', () => {
-            resolve();
-          })
-          .on('close', () => {
-            resolve();
-          });
+        console.log(this.executeTask.length);
```
Comment (dzearing) [inline src/GulpTask.ts]: console.log

**Comment by harry8525** [comment]
Fixed a few more bugs and handled your pr feedback

**Comment by dzearing** [comment]
Approved

**Pull request merged**